Add 0.333.0 release notes (#526)
trinodb/trino-python-client: CHANGES.md

[README](https://github.com/trinodb/trino-python-client/blob/master/README.md)
and the [PyPI page](https://pypi.org/project/trino/).

## Release 0.333.0

> * Improve handling of query results containing null error field.

Reviewer: ```suggestion
* Improve handling of query results containing `null` values for the error field.
```

  ([#512](https://github.com/trinodb/trino-python-client/pull/512))

> * Fix accessing spooled segments on S3 when authentication to Trino cluster

Reviewer: ```suggestion
* Fix accessing spooled segments on S3 when authentication to the Trino cluster
```

  is used.
  ([#525](https://github.com/trinodb/trino-python-client/pull/525))

> * Simplify `segment` cursor API, that makes it easier to iterate over spooled

Reviewer: ```suggestion
* Simplify `segment` cursor API to make it easier to iterate over spooled
```
